massgov/mayflower
# React: Add Standard Accordion Components to Mayflower React

**Description:** This PR includes the addition of 4 new components that together create a standard accordion feature in Mayflower React. The design is pulled from Mayflower PL but were refactored in order to generate these new components. Also, in order to enable animations, this PR installs the react-transition-group library.

**Question to Validate with React Team** --> Is the approach to handling children in both accordion item and accordion wrapper appropriate here? Is the approach to animations here appropriate? Do we want to pull animation and focus scss in accordion item out to a higher level in the react library so it applies to all relevant components?

External Review Steps:
1. npm install
2. Go to @atoms/icons/SvgCircleChevron and confirm that the icon exist and matches [mayflower PL](https://mayflower.digital.mass.gov/?p=atoms-svg-circle-chevron) 
3. Go to @molecules/AccordionItem and confirm that a single accordion item is displayed, matching the appearance of a single item in [Mayflower PL](https://mayflower.digital.mass.gov/?p=organisms-collapsible-content-as-steps)
4. Go to @organisms/AccordionWrapper and confirm that multiple accordion items are grouped and follow the same behavior as [Mayflower PL](https://mayflower.digital.mass.gov/?p=organisms-collapsible-content-as-steps) 


Updated Gifs
![accordionnoicon](https://user-images.githubusercontent.com/9359261/44352702-1082c500-a473-11e8-876b-8a9ba8cac3b9.gif)
![accordionsecondary](https://user-images.githubusercontent.com/9359261/44352703-1082c500-a473-11e8-96e7-d2f63a671093.gif)
![accordiondefault](https://user-images.githubusercontent.com/9359261/44352704-1082c500-a473-11e8-99bd-beb6061df50c.gif)
